dusty/compiler/spec_assembler.py

```diff
@@ -93,6 +93,18 @@ def _filter_active(spec_type, specs):
             del specs[spec_type][item_name]
     logging.info("Spec Assembler: filtered active {} to {}".format(spec_type, set(specs[spec_type].keys())))
 
+def _add_active_assets(specs):
+    specs['assets'] = {}
+    for spec in specs.get_apps_and_libs():
+        for asset in spec['assets']:
+            if not specs['assets'].get(asset['name']):
+                specs['assets'][asset['name']] = {}
+                specs['assets'][asset['name']]['required_by'] = set()
+                specs['assets'][asset['name']]['used_by'] = set()
+            specs['assets'][asset['name']]['used_by'].add(spec.name)
+            if asset['required']:
+                specs['assets'][asset['name']]['required_by'].add(spec.name)
```

Comment:
I'm wondering if instead of this function, I should expand assets inside apps specs.  So that an app spec will have all the assets that it and all of its libs need in the same place